libs/filamat/src/MaterialBuilder.cpp

```diff
@@ -486,14 +488,20 @@ Package MaterialBuilder::build() noexcept {
                     errorOccured = true;
                     break;
                 }
+
                 if (targetApi == TargetApi::OPENGL) {
+                    if (postProcessor.hasSpirvOptimization()) {
```

Comment:
We don't need a new `hasSpirvOptimization` method, you can use the `codeGenTargetApi` variable which is set via line 63:

```
    // OpenGL is a special case. If we're doing any optimization, then we need to go to Spir-V.
    TargetApi glCodeGenTargetApi = mOptimization > MaterialBuilder::Optimization::PREPROCESSOR ?
            TargetApi::VULKAN : TargetApi::OPENGL;
```

Then the condition would be:

```
if (codeGenTargetApi == TargetApi::VULKAN)
```